### PR TITLE
chore: upgrade to AGP 8.0

### PR DESCRIPTION
**Pull request by thatfiredev**
This PR should:
- Upgrade the project to AGP 8.0
- Update Github Actions to use Java 17
- add extra jvm args for ktlint to run in Java 17
- Change the compile target from java 1.8 to java 11

### CODE REVIEW

File: admob/app/build.gradle [hunk 1]
```diff
@@ -32,8 +32,13 @@ android {
...
     }
-
-
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline admob/app/build.gradle:41]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: analytics/app/build.gradle [hunk 2]
```diff
@@ -27,6 +27,14 @@ android {
...
     }
 
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+
```
Comment (thatfiredev) [inline analytics/app/build.gradle:37]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }

```

File: appdistribution/app/build.gradle [hunk 3]
```diff
@@ -24,7 +24,13 @@ android {
...
         }
     }
-
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline appdistribution/app/build.gradle:33]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: auth/app/build.gradle [hunk 4]
```diff
@@ -32,6 +32,9 @@ android {
         sourceCompatibility 1.8
         targetCompatibility 1.8
     }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline auth/app/build.gradle:37]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: build.gradle [hunk 5]
```diff
@@ -5,7 +5,7 @@ buildscript {
         mavenLocal()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.4.2'
+        classpath 'com.android.tools.build:gradle:8.0.0-rc01'
```
Comment (thatfiredev) [inline build.gradle:8]: It's stable now
```suggestion
        classpath 'com.android.tools.build:gradle:8.0.0'
```

File: config/app/build.gradle [hunk 6]
```diff
@@ -26,6 +26,13 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline config/app/build.gradle:34]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: crash/app/build.gradle [hunk 7]
```diff
@@ -31,6 +31,13 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline crash/app/build.gradle:39]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: database/app/build.gradle [hunk 8]
```diff
@@ -27,6 +27,13 @@ android {
             signingConfig signingConfigs.debug
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline database/app/build.gradle:35]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: dynamiclinks/app/build.gradle [hunk 9]
```diff
@@ -39,6 +39,13 @@ android {
             resValue "string", "dynamic_links_uri_prefix", "https://YOUR_APP.page.link"
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline dynamiclinks/app/build.gradle:47]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: firestore/app/build.gradle [hunk 10]
```diff
@@ -34,6 +34,13 @@ android {
             signingConfig signingConfigs.debug
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline firestore/app/build.gradle:43]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: functions/app/build.gradle [hunk 11]
```diff
@@ -28,6 +28,13 @@ android {
             signingConfig signingConfigs.debug
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline functions/app/build.gradle:37]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 12]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```
Comment (thatfiredev) [inline gradle/wrapper/gradle-wrapper.properties:3]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.2-bin.zip
```

File: inappmessaging/app/build.gradle [hunk 13]
```diff
@@ -24,7 +24,13 @@ android {
...
         }
     }
-
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
```
Comment (thatfiredev) [inline inappmessaging/app/build.gradle:33]:
```suggestion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
    }
```

File: messaging/app/build.gradle [hunk 14]
```diff
@@ -32,6 +32,13 @@ android {
             excludes += ['LICENSE.txt']
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline messaging/app/build.gradle:40]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: perf/app/build.gradle [hunk 15]
```diff
@@ -33,6 +33,13 @@ android {
             }
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline perf/app/build.gradle:41]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

File: storage/app/build.gradle [hunk 16]
```diff
@@ -25,6 +25,13 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (thatfiredev) [inline storage/app/build.gradle:33]:
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = '11'
```

**Review by marinacoelho — APPROVED** [review]

**Review by Kryslynn93 — COMMENTED** [review]
Ol

**Pull request merged**